designs/mutually-exclusive-pods-uniformly-distributed-scheduling.md

```diff
@@ -0,0 +1,55 @@
+# Mutually Exclusive Pods Uniformly Distributed Scheduling
+
+## Background
+
+> [issues 1418](https://github.com/kubernetes-sigs/karpenter/issues/1418)
+
+I have 10 pending pods:
+
+- **pod1**: 1c1g requests, with anti-affinity; cannot be scheduled on the same node as pod10 and pod9.
+- **pod2 ~ pod8**: 1c1g requests; no anti-affinity is configured.
+- **pod9**: 1c1g requests, with anti-affinity; cannot be scheduled on the same node as pod1 and pod10.
+- **pod10**: 1c1g requests, with anti-affinity; cannot be scheduled on the same node as pod1 and pod9.
+
+In Karpenter, it creates three nodes, with the first node being super large and the remaining two being very small:
+
+- **node1**: c7a.4xlarge, 16c32g (8Pods)
+- **node2**: c7a.large, 2c4g (1Pod)
+- **node3**: c7a.large, 2c4g (1Pod)
+
+**Expected Behavior**:
+I want the resources of the three nodes to be evenly distributed, like:
+
+- **node1**: c7a.4xlarge, 8c16g (4Pod)
+- **node2**: c7a.xlarge, 4c8g (3Pod)
+- **node3**: c7a.xlarge, 4c8g (3Pod)
+
+In this situation, the cluster will be more stable (e.g., draining one node will not cause most pods to be rescheduled).
+
+## Recommended Solution
+
+By sorting in byCPUAndMemoryDescending, pods with the same specifications (cpu or memory request equal) and mutually exclusive pods (with PodAntiAffinity or TopologySpreadConstraints) are prioritized to be scheduled first.
+
+Code implementation: [pull 1548](https://github.com/kubernetes-sigs/karpenter/pull/1548)
+
+### Test Results
+
```

Comment:
What will be the impact if a user upgrades to a version that includes this feature? 

Will the existing nodes need to be recreated according to the new logic?

Reply:
The new scheduling results are more balanced, and that's about it. No changes are needed for that.